src/openms/include/OpenMS/CHEMISTRY/Residue.h

```diff
@@ -393,6 +397,10 @@ namespace OpenMS
     /// helper for mapping residue types to letters for Text annotations and labels
     static char residueTypeToIonLetter(const ResidueType& res_type);
 
+    /// Write as Origin+Modification, e.g. M(Oxidation), or X[945.34] or N[+14.54] for user-defined mods.
+    /// This requires the Residue to have a valid OneLetterCode and an optional (but valid) ResidueModification (see ResidueModification::toString())
+    String toSequence() const;
```

Comment:
toString ?

Reply:
I thought about that too, but I went for `toSequence` to make clear that its not just any string, but a proper sequence, which can be parsed again

Reply:
hmm I don't like the name sequence for that... we have toString() and toUnmodifiedString() for AASequence
but never used sequence

Reply:
alternatively to toString()
is there something we already use?
Also, note that we have on the Modification site something related (but different) string
M(Oxidation) vs Oxidation (M)
Would be good to have more telling names here in general